src/transforms/coercer.rs

```diff
@@ -60,7 +60,9 @@ impl Transform for Coercer {
             for (field, conv) in &self.types {
                 if let Some(value) = log.remove(field) {
                     match conv.convert(value) {
-                        Ok(converted) => new_log.insert(field, converted),
+                        Ok(converted) => {
+                            new_log.insert(field, converted);
+                        }
```

Comment:
These (and other related changes in your patch) could be reduced to a single line with `drop(new_log.insert(field, converted))` which avoids the need for a block statement. This may not be recommended though due to the confusion that the `drop` may add.

Reply:
Kind of how I felt too. :S